docs(main): update proxy doc (#3747)
web-infra-dev/modern.js: packages/document/main-doc/docs/en/components/global-proxy.mdx

Modern.js provides an out-of-the-box global proxy plugin `@modern-js/plugin-proxy`, which is based on [whistle](https://github.com/avwo/whistle) and can be used to view and modify the requests and responses of HTTP/HTTPS, as well as be used as an HTTP proxy server.

### Setting Proxy Rules

> After installing the proxy plugin and filling in the rules, execute `pnpm run dev`. Modern.js will automatically enable the proxy server after the development server starts.

Reviewer: ```suggestion
After installing the proxy plugin and configuring the rules, run `pnpm run dev`. Modern.js will automatically enable the proxy server when the development server is started.
```

Specific proxy rules can be set via the [`dev.proxy`](/configure/app/dev/proxy) or the `config/proxy.js` file.

### Proxy Dashboard

> After installing the proxy plugin and configuring the proxy rules, execute the `pnpm run dev` command:

Reviewer: ```suggestion
After installing the proxy plugin and configuring the proxy rules, run the `pnpm run dev` command:
```
